### PR TITLE
Toggle from child card to parent card

### PR DESCRIPTION
**Pull request by sidkotha**
## Description
Toggle Visibility from child card to Main Card does not work

## Sample Card 1 - when view identifier available in parent card
`{
  "type": "AdaptiveCard",
  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
  "version": "1.2",
  "body": [
    {
      "type": "ColumnSet",
      "columns": [
        {
          "type": "Column",
          "width": "stretch",
          "items": [
            {
              "type": "TextBlock",
              "text": "Column 1",
              "wrap": true
            }
          ],
          "style": "attention",
          "id": "column1"
        },
        {
          "type": "Column",
          "width": "stretch",
          "items": [
            {
              "type": "TextBlock",
              "text": "Column 2",
              "wrap": true
            }
          ],
          "style": "attention",
          "id": "column2"
        }
      ],
      "style": "good",
      "id": "columnSet"
    }
  ],
  "actions": [
    {
      "type": "Action.ToggleVisibility",
      "title": "Column Set",
      "targetElements": [
        "columnSet"
      ]
    },
    {
      "type": "Action.ToggleVisibility",
      "title": "Column 1",
      "targetElements": [
        "column1"
      ]
    },
    {
      "type": "Action.ToggleVisibility",
      "title": "Column 2",
      "targetElements": [
        "column2"
      ]
    },
    {
      "type": "Action.ShowCard",
      "title": "Action.ShowCard",
      "card": {
        "type": "AdaptiveCard",
        "body": [
          {
            "type": "ActionSet",
            "actions": [
              {
                "type": "Action.ToggleVisibility",
                "title": "Column Set",
                "targetElements": [
                  "columnSet"
                ]
              },
              {
                "type": "Action.ToggleVisibility",
                "title": "Column 1",
                "targetElements": [
                  "column1"
                ]
              },
              {
                "type": "Action.ToggleVisibility",
                "title": "Column 2",
                "targetElements": [
                  "column2"
                ]
              },
              {
                "type": "Action.ShowCard",
                "title": "Action.ShowCard",
                "card": {
                  "type": "AdaptiveCard",
                  "body": [
                    {
                      "type": "ActionSet",
                      "actions": [
                        {
                          "type": "Action.ToggleVisibility",
                          "title": "Column Set",
                          "targetElements": [
                            "columnSet"
                          ]
                        },
                        {
                          "type": "Action.ToggleVisibility",
                          "title": "Column 1",
                          "targetElements": [
                            "column1"
                          ]
                        },
                        {
                          "type": "Action.ToggleVisibility",
                          "title": "Column 2",
                          "targetElements": [
                            "column2"
                          ]
                        },
                        {
                          "type": "Action.ShowCard",
                          "title": "Action.ShowCard",
                          "card": {
                            "type": "AdaptiveCard",
                            "body": [
                              {
                                "type": "ActionSet",
                                "actions": [
                                  {
                                    "type": "Action.ToggleVisibility",
                                    "title": "Column Set",
                                    "targetElements": [
                                      "columnSet"
                                    ]
                                  },
                                  {
                                    "type": "Action.ToggleVisibility",
                                    "title": "Column 1",
                                    "targetElements": [
                                      "column1"
                                    ]
                                  },
                                  {
                                    "type": "Action.ToggleVisibility",
                                    "title": "Column 2",
                                    "targetElements": [
                                      "column2"
                                    ]
                                  }
                                ]
                              }
                            ]
                          },
                          "id": "showCard3"
                        }
                      ]
                    }
                  ]
                },
                "id": "showCard2"
              }
            ]
          }
        ]
      },
      "id": "showCard"
    }
  ]
}`
Toggle Visibilility Show Card to Main Card - 3

## Sample card -2  when view identifier not available
`{
   "$schema":"http://adaptivecards.io/schemas/adaptive-card.json",
   "type":"AdaptiveCard",
   "version":"1.2",
   "body":[
      {
         "type":"TextBlock",
         "text":"Press the buttons to toggle the images!",
         "wrap":true
      },
      {
         "type":"TextBlock",
         "text":"Here are some images:",
         "isVisible":false,
         "id":"textToToggle"
      },
      {
         "type":"ColumnSet",
         "columns":[
            {
               "type":"Column",
               "items":[
                  {
                     "type":"Image",
                     "url":"https://adaptivecards.io/content/cats/1.png",
                     "isVisible":false,
                     "id":"imageToToggle2",
                     "altText":"sample image 2",
                     "size":"medium"
                  }
               ],
               "width":"auto"
            }
         ]
      }
   ],
   "actions":[
      {
         "type":"Action.ToggleVisibility",
         "title":"Toggle!",
         "targetElements":[
            "textToToggle",
            "imageToToggle2"
         ]
      },
{
                "type": "Action.ToggleVisibility",
                "title": "Column 2",
                "targetElements": [
                  "column2"
                ]
              },
              {
                "type": "Action.ShowCard",
                "title": "ShowCard2",
                "card": {
                  "type": "AdaptiveCard",
                  "body": [
                    {
                      "type": "ActionSet",
                      "actions": [
{
                "type": "Action.ToggleVisibility",
                "title": "Column 2",
                "targetElements": [
                  "column2"
                ]
              },
              {
         "type":"Action.ToggleVisibility",
         "title":"Toggle!",
         "targetElements":[
            "textToToggle",
            "imageToToggle2"
         ]
      }
                      ]
                    }
                  ]
                },
                "id": "showCard2"
              }
   ]
}`

### CODE REVIEW

File: source/macos/AdaptiveCards/AdaptiveCards/Views/ACRContentStackView.swift [hunk 1]
```diff
@@ -215,37 +215,36 @@ class ACRContentStackView: NSView, ACRContentHoldingViewProtocol, SelectActionHa
...
-                    for index in toggledViewIndex + 1..<stackView.arrangedSubviews.count {
-                        if let nextView = stackView.arrangedSubviews[index] as? ACRContentStackView, !nextView.isHidden {
-                            nextView.currentSpacingView?.isHidden = false
-                            nextView.currentSeparatorView?.isHidden = false
-                        }
-                    }
-                }
-            }
-        }
+             // if an object is hidden then we need to update the visibility of spacing view of the adjacent nonHidden object
```
Comment (sidkotha) [inline source/macos/AdaptiveCards/AdaptiveCards/Views/ACRContentStackView.swift:218]: this file change is just a refactor where earlier "if" conditions were there ,now we have "guard" statements used.

**Review by mukuagar — APPROVED** [review]
Changes look good to me, just want to confirm if the recursion doesn't go into a loop for any case though

File: source/macos/AdaptiveCards/AdaptiveCardsTests/AdaptiveCardsTests.swift [hunk 2]
```diff
@@ -69,6 +69,54 @@ class AdaptiveCardsTests: XCTestCase {
...
+        let bundle = Bundle(for: type(of: self))
+        let filename = "ToggleVisibilityChildToParent.json"
+        
+        let dataAsset = NSDataAsset(name: filename, bundle: bundle)?.data
+        guard let data = dataAsset else { return }
+        let jsonString = String(data: data, encoding: .utf8)!
+        
+        let hostConfigData = NSDataAsset(name: "HostConfig.json", bundle: bundle)!.data
+        let hostConfigJSON = String(data: hostConfigData, encoding: .utf8)!
+        let hostConfig = try! AdaptiveCard.parseHostConfig(from: hostConfigJSON).get()
```
Comment (mukuagar) [inline source/macos/AdaptiveCards/AdaptiveCardsTests/AdaptiveCardsTests.swift:82]: Very nice test!

**Comment by sidkotha** [comment]
> Changes look good to me, just want to confirm if the recursion doesn't go into a loop for any case though

yes verifying only that with different scenarios.

**Pull request merged**